blockstream.go

```diff
@@ -97,8 +101,15 @@ func (bs *BlockStream) Session(ctx context.Context, token access.Token, autosave
 			defer bs.wg.Done()
 
 			if err := f(); err != nil {
-				log.Error(err)
 				s.Reset()
+				ses.removeProvider()
+
+				log.Error(err)
+				if ses.prvs == 0 {
+					log.Error("Closing session: ", ErrStreamsReset)
+					ses.cancel()
+					ses.cherr <- ErrStreamsReset
```

Comment:
Seems like the chan is always nil.

Reply:
why? it gets an error from `requster.writeLoop` and `requester.writeLoop`